core/src/transport.rs

```diff
@@ -152,9 +153,8 @@ pub trait Transport {
     /// Boxes the transport, including custom transport errors.
     fn boxed(self) -> boxed::Boxed<Self::Output>
     where
-        Self: Transport + Sized + Send + Sync + 'static,
+        Self: Transport + Sized + Send + Sync + Unpin + 'static,
```

Comment:
I did not manage to implement the polling of the inner transport without it being `Unpin`.  Happy for any suggestions on how to solve this without the trait bound.

Reply:
Whilst I understanding pinning in principle, I find it tricky to wrap my head around it every time. I am wondering though: We are _moving_ self here and constructing a trait object. I think because of the type erasure, it is not safe to move this type around unless we can guarantee that it is `Unpin`. Thus requiring this trait bound makes sense I think.